### PR TITLE
Decompile Bartoli

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [X] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

#1691

### CODE REVIEW

File: src/tr2/game/objects/creatures/bartoli.c [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    item->pos.z -= 2 * STEP_L;
+
+    const int16_t item_dragon_back_num = Item_Create();
+    const int16_t item_dragon_front_num = Item_Create();
+    if (item_dragon_back_num == NO_ITEM || item_dragon_front_num == NO_ITEM) {
+        Shell_ExitSystem("Unable to create dragon");
+    }
+
+    ITEM *const item_dragon_back = Item_Get(item_dragon_back_num);
+    item_dragon_back->object_id = 23;
```
Comment (lahm86) [inline src/tr2/game/objects/creatures/bartoli.c]: `= O_DRAGON_BACK`

**Review by lahm86 — APPROVED** [review]
Looks great, just a missing enum usage.

File: src/tr2/game/objects/creatures/bartoli.c (hunk 1 above)
Comment (rr-) [reply src/tr2/game/objects/creatures/bartoli.c]: Thanks, fixed

**Comment by aredfan** [comment]
I found these issues.

- [x] The twinkles will stop after a short duration when the dragon is down.

Left is OG, right is PR.



- [x] If the dragon gets up again, fire and explosion sprites will stop being visible. Lara doesn't catch fire.

**Comment by rr-** [comment]
@aredfan thanks. Can you please check now?

**Review by aredfan — APPROVED** [review]
@rr- no problem. 👍 

I ran another test and both issues are resolved.

**Pull request merged**